samples/sample-actions-bar-plugin/tests/test.spec.ts

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { test as base, expect, TestInfo } from '@playwright/test';
+import { secret, server } from '../../../tests/core/parameters';
+import { Sample } from '../../../tests/core/sample';
+import { encodeCustomParams } from '../../../tests/core/helpers';
+import { elements as e } from './elements';
+import { extractObject } from './utils/extractObject';
+
+let pluginUrl: string | undefined = process.env.ACTIONS_BAR_URL;
+
+const test = base.extend<{
+  sampleTest: Sample;
+}>({
+  sampleTest: async ({ browser, context, page }, use) => {
+    if (!pluginUrl) throw new Error('Plugin URL is not set from beforeAll.');
+    const createParameter = encodeCustomParams(`pluginManifests=${JSON.stringify([{ url: pluginUrl }])}`);
+    const sampleTest = new Sample({ browser, context });
+    await sampleTest.initModPage(page, { createParameter });
+    await use(sampleTest);
+  },
+});
+
+test.describe.parallel('Action bar', () => {
+  test.beforeAll(async ({ request }, testInfo: TestInfo) => {
+    if (!server) return test.skip(true, 'No server URL variable provided. Skipping test');
+    if (!secret) return test.skip(true, 'No server secret variable provided. Skipping test');
+
+    const serverDomain = new URL(server).origin;
+    const manifestUrlPath = '/plugins/sample-actions-bar-plugin/dist/manifest.json';
+    pluginUrl = `${serverDomain}${manifestUrlPath}`;
+
+    const response = await request.get(pluginUrl);
+    test.skip(!response.ok(), `Failed to fetch plugin manifest for ${testInfo.title} plugin. returned status ${response.status()}`);
+
+    try {
+      return response.json();
+    } catch (error) {
+      return test.skip(error, `Invalid JSON response from plugin manifest for ${testInfo.title} plugin`);
+    }
+  });
+
+  test('should log expected message when custom SVG button is clicked', async ({ sampleTest }) => {
+    await sampleTest.modPage.hasElement(e.actionsBarButtonCustomSvg, 'should display the button with custom SVG');
+    const [consoleMessage] = await Promise.all([
+      sampleTest.modPage.page.waitForEvent('console'),
```

Comment:
I had an error that the console message I got was

> Warning: findDOMNode is deprecated and will be removed in the next major release. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-find-node

Which is a message that commonly appear in bbb.

One way to work around that is to filter the console logs by the pluginLogger (as this is emitted by our custom plugin logger)

Reply:
This happened to me for `sample-actions-bar-plugin` in 
- should display the expected log when the button with custom SVG is clicked;
- should log the expected string when the toggle group button is clicked

And it's not consistent.

Reply:
I've tried to hit this bug locally in many ways, but could not even once. However, the failure is reasonable and likely to happen. dbfb70f fix it, as it filters by `PluginLogger` as suggested